doc/design/bucketpath.rst

```diff
@@ -0,0 +1,157 @@
+==============================
+Design Document Bucket Path(s)
+==============================
+
+Introduction
+============
+
+- Purpose of the document / Problem description
+- Definitions and Acronyms
+
+Overview
+========
+
+Design Goals
+============
+
+Path Design
+===========
+
+Future Considerations
+=====================
+
+Sudo Code Outline
+=================
+
+.. code-block:: python
+
+    from exasol.bucketfs import PathBuilder
+
+    Path = PathBuilder(credentials)
+
+    udf_path = Path("bfsl://some/local/path/file.tar.gz")
+    http_bucket_path = Path("bfs://127.0.0.1:8888/service/bucket/some/file.tar.gz")
+    https_bucket_path = Path("bfss://127.0.0.1:8888/service/bucket/some/file.tar.gz")
```

Comment:
I am actually not sure if we should point to files at all, because Bucket and what ever we define for DirectoryBucket can also only point to a Directory. Except, we want to go futher and actually point to a File Abstraction.

Reply:
We don't need to but it should be possible, I think `Path` is the right abstraction.
Given that we can discuss on it's feature set.

Reply:
Yeah, after reading the Path class, it probably makes sense to point to files. If I think about the usage, we actually might want to point to a file, such that a UDF uses a specific file for example.